daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/SBuiltin.scala

```diff
@@ -952,12 +952,15 @@ private[lf] object SBuiltin {
           stakeholders = cached.stakeholders,
           key = cached.key,
           version = machine.tmplId2TxVersion(cached.templateId),
-        )
-
-      onLedger.updateCachedContracts(coid, cached)
-      onLedger.ptx = newPtx
-      checkAborted(onLedger.ptx)
-      machine.returnValue = SContractId(coid)
+        ) match {
+        case Left((newPtx, err)) =>
```

Comment:
Conventionally, the successful cases should be matched first. This is a general principal, since scala pattern matching is linear we favor the successful cases, that we assume to be more frequent than error cases.

Reply:
did this